crev-common/src/lib.rs

```diff
@@ -48,6 +48,49 @@ pub fn base64_encode<T: ?Sized + AsRef<[u8]>>(input: &T) -> String {
     base64::encode_config(input, base64::URL_SAFE_NO_PAD)
 }
 
+/// Takes an identifier like "https://crates.io" and converts it to something safe for use in paths etc.
+/// 
+/// # Examples
+/// 
+/// ```
+/// # use std::path::Path;
+/// # use crev_common::sanitize_name;
+/// // Pass through when able
+/// assert_eq!(sanitize_name("lazy_static"), Path::new("lazy_static"));
+/// 
+/// // Hash reserved windows filenames (or any other 3 letter name)
```

Comment:
Oh. Nice. I wouldn't have though of that. :)

Reply:
I'm always particularly paranoid of that after someone uploaded a crate named `nul` and broke crates.io for a bit 😆 : https://www.reddit.com/r/rust/comments/68hemz/i_think_a_crate_called_nul_is_causing_errors_for/

But apparently not paranoid enough...  as there's also some 4 letter reserved names I forgot about!  Fortunately hashing unconditionally will fix that...

Reply:
Ha. :smile: